packages/component-base/src/overflow-controller.d.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright (c) 2021 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import type { ReactiveController } from 'lit';
+
+/**
+ * A controller that detects if content inside the element overflows its scrolling viewport,
+ * and sets the `overflow` attribute on the host with a value that indicates the directions
+ * where content is overflowing. Supported values are: `top`, `bottom`, `start`, `end`.
+ */
+export class OverflowController implements ReactiveController {
```

Comment:
I'm getting a Typescript error here, it seems the class must implement at least one of the optional methods in `ReactiveController`.

Reply:
Thanks, updated to use `hostConnected` as we do elsewhere, e.g. in `SlotController`.